lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -226,20 +247,13 @@ private void clearOauthSessionIfExists(IpvSessionItem ipvSessionItem) {
     private void updateUserSessionForTimeout(String oldState, IpvSessionItem ipvSessionItem) {
         ipvSessionItem.setErrorCode(OAuth2Error.ACCESS_DENIED.getCode());
         ipvSessionItem.setErrorDescription(OAuth2Error.ACCESS_DENIED.getDescription());
-        ipvSessionItem.setUserState(CORE_SESSION_TIMEOUT_STATE);
-        ipvSessionService.updateIpvSession(ipvSessionItem);
-        var message =
-                new StringMapMessage()
-                        .with("journeyEngine", "State transition")
-                        .with("event", "timeout")
-                        .with("from", oldState)
-                        .with("to", CORE_SESSION_TIMEOUT_STATE);
-        LOGGER.info(message);
+        ipvSessionItem.setJourneyType(IpvJourneyTypes.SESSION_TIMEOUT);
+        updateUserState(oldState, CORE_SESSION_TIMEOUT_STATE, "timeout", ipvSessionItem);
     }
 
     @Tracing
     private boolean sessionIsNewlyExpired(IpvSessionItem ipvSessionItem) {
-        return (!CORE_SESSION_TIMEOUT_STATE.equals(ipvSessionItem.getUserState()))
+        return (!IpvJourneyTypes.SESSION_TIMEOUT.equals(ipvSessionItem.getJourneyType()))
```

Comment:
This is checking whether you are (anywhere) within the 'timeout' journey, rather than a specific state. So e.g. if we added more screens in the timeout journey it would still work.

It's also nicer to rely on the enum, rather than a hardcoded string (that needs to match the YAML).